Structure_Engine/Create/Elements/RetainingWall.cs

```diff
@@ -0,0 +1,116 @@
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SurfaceProperties;
+using BH.Engine.Base;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.Engine.Structure;
+
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a RetainingWall from a Stem, Footing and properties.")]
+        [Input("stem", "The Stem of the RetainingWall.")]
+        [Input("footing", "The Footing of the RetainingWall.")]
+        [Input("retainedHeight", "The retained height of soil measured from the bottom of the wall Footing.")]
+        [Input("coverDepth", "The distance from top of Footing to finished floor level on the exposed face.")]
+        [Input("groundWaterDepth", "The distance from the base of the Footing to ground water level.")]
+        [Input("retentionAngle", "A property of the material being retained measured from the horizontal plane.")]
+        [Output("retainingWall", "The created RetainingWall containing the stem and footing.")]
+        public static RetainingWall RetainingWall(Stem stem, PadFoundation footing, double retainedHeight, double coverDepth, double retentionAngle, double groundWaterDepth = 0)
+        {
+            if (stem.IsNull() || footing.IsNull())
+                return null;
+
+            if (Query.IsValid(stem, footing))
+                return new RetainingWall() { Stem = stem, Footing = footing, RetainedHeight = retainedHeight, CoverDepth = coverDepth, RetentionAngle = retentionAngle, GroundWaterDepth = groundWaterDepth };
+            else
+                return null;
+        }
+
+        /***************************************************/
+
+        //TODO Add input descriptions here
+        [Description("Create RetainingWall from a Line and defining properties.")]
+        [Output("retainingWall", "RetainingWall with specified properties")]
+        public static RetainingWall RetainingWall(
+            Line line,
+            double retainedHeight,
+            Vector normal,
+            double stemThickness,
+            double toeLength,
+            double heelLength,
+            ConstantThickness footingThickness,
+            double coverDepth,
+            double retentionAngle
+            )
+        {
+            if (line.Start == null && line.End == null) { return null; }
+
+            if (!line.IsInPlane(Geometry.Create.Plane(line.Start, Vector.ZAxis)))
+            {
+                Base.Compute.RecordError("Provided line is not parallell to the XY plane. Please provide a line parallell to the XY plane.");
```

Comment:
```suggestion
                Base.Compute.RecordError("Provided line is not within a single Plane. Please provide a line that exists within in a single plane.");
```

Reply:
Stated in descriptions that it should be parallell to XY plane